code/modules/mob/living/carbon/human/human_helpers.dm

```diff
@@ -85,11 +85,27 @@
 		. = if_no_id	//to prevent null-names making the mob unclickable
 	return
 
-//gets ID card object from special clothes slot or null.
-/mob/living/carbon/human/get_idcard()
-	if(wear_id)
+//Gets ID card from a human. If hand_first is false the one in the id slot is prioritized, otherwise inventory slots go first.
+/mob/living/carbon/human/get_idcard(hand_first = TRUE)
+	//Check hands
+	var/obj/item/I = get_active_held_item()
+	if(!I)
+		I = get_inactive_held_item()

```

Comment:
If your inactive hand holds an id and your active hand holds anything else this wont work